dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -1,820 +1,397 @@
-
-**Design for consistency**
-
-If your API needs to support multiple data types for the same logical concept, consider these approaches:
-
-1. **Use separate, clearly named fields**
-   ```json
-   {
-     "group_by_field": "field_name",      // Single field
-     "group_by_fields": ["field1", "field2"]  // Multiple fields
-   }
-   ```
-
-2. **Use the most flexible type from the start**
-   ```json
-   {
-     "group_by": ["field_name"]  // Always an array, even for single values
-   }
-   ```
-
-**When you have to support flexible types**
-
-If polymorphic behavior is unavoidable, JSON strings provide a workaround—but use them sparingly:
-
-```terraform
-resource "elasticstack_elasticsearch_index" "example" {
-  # Structured fields for simple, predictable values
-  number_of_shards   = 1
-  number_of_replicas = 2
-  # JSON strings only for truly complex, variable structures
-  mappings = jsonencode({
-    properties = {
-      user = { type = "keyword" }
-      timestamp = { type = "date" }
-    }
-  })
-}
-```
+If you are in doubt, rather go with stricter validation. Make a requirement more lax if needed is never a breaking change!
 
-[Index resource schema](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/elasticsearch/index/schema.go)
+### Headers
 
-**What's the issue with JSON strings?**
+**Should not be used to specify behavior**
 
-`jsonencode()` and `jsondecode()` come with significant downsides:
+Outside of exceptional cases, you should always use the parameters, query parameters or body of the request to specify behavior.
 
-- **Limited validation**: Terraform can only validate JSON syntax during plan, not the actual content structure
-- **Poor user experience**: Users lose autocompletion, type checking, and clear error messages
-- **Complex validation logic**: Content validation is left to the API layer, leading to runtime errors that could have been plan-time validation
+### Response codes
 
-**Bottom line**
+**Should always be used as close as possible to their semantic meaning**
 
-Consistent field types make everyone's life easier—API consumers, Terraform provider developers, and your future self. When designing your API, choose field types that can accommodate future needs rather than changing them later.
+See the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status).
 
-### Make requests transactional
+### Pagination and sorting
 
-Terraform expects changes to take effect immediately. Design your API so that changes take effect immediately upon successful response.
+**Page-based request**
 
-Consider the scenario:
+`GET /api/resources?page=1&size=10`
 
-A user wants to create alerting rules with different configurations, depending on the space they are in.
+**Page-based response**
 
-In Kibana’s UI, they’d create a space, change into that space and then create the rule.
+```json
+{
+  "items": [...],
+  "total": 100,
+  "page": 1,
```

Comment:
Alternatively we can use `from` as this provides a more flexible form of pagination. E.g. Get pageNumber of resources "from" 22nd entry.